packages/Meshfree/src/DTK_DetailsMovingLeastSquaresOperatorImpl.hpp

```diff
@@ -0,0 +1,305 @@
+                                                   source_points.extent( 0 ) );
+
+        Kokkos::parallel_for(
+            DTK_MARK_REGION( "compute_radius" ),
+            Kokkos::RangePolicy<ExecutionSpace>( 0, n_target_points ),
+            KOKKOS_LAMBDA( int const i ) {
+                double distance = 10. * Kokkos::ArithTraits<double>::epsilon();
+                for ( int j = offset( i ); j < offset( i + 1 ); ++j )
+                {
+                    double new_distance = Details::distance(
+                        Point{{source_points( j, 0 ), source_points( j, 1 ),
+                               source_points( j, 2 )}},
+                        Point{{0., 0., 0.}} );
+
+                    if ( new_distance > distance )
+                        distance = new_distance;
+                }
+                // If a point is exactly on the boundary of the compact domain,
+                // its weight will be zero so we need to make sure that no point
+                // is exactly on the boundary.
+                for ( int j = offset( i ); j < offset( i + 1 ); ++j )
+                    radius( j ) = 1.1 * distance;
+            } );
+
+        return radius;
+    }
+
+    // We need the third argument because otherwise the compiler cannot do the
+    // template deduction. For some unknown raison, explicitly chosing the
+    // value of the template parameter does not work.
+    template <typename RBF>
+    static Kokkos::View<double *, DeviceType>
+    computeWeights( Kokkos::View<double const **, DeviceType> source_points,
+                    Kokkos::View<double const *, DeviceType> radius,
+                    RBF const & )
+    {
+        auto const n_source_points = source_points.extent( 0 );
+
+        int const spatial_dim = 3;
+        DTK_REQUIRE( source_points.extent_int( 1 ) == spatial_dim );
+
+        Kokkos::View<double *, DeviceType> phi( "weights", n_source_points );
+        Kokkos::parallel_for(
+            DTK_MARK_REGION( "compute_weights" ),
+            Kokkos::RangePolicy<ExecutionSpace>( 0, n_source_points ),
+            KOKKOS_LAMBDA( int i ) {
+                RadialBasisFunction<RBF> rbf( radius( i ) );
+                phi( i ) = rbf( Details::distance(
+                    Point{{source_points( i, 0 ), source_points( i, 1 ),
+                           source_points( i, 2 )}},
+                    Point{{0., 0., 0.}} ) );
+            } );
+        Kokkos::fence();
+        return phi;
+    }
+
+    template <typename PolynomialBasis>
+    static Kokkos::View<double *, DeviceType>
+    computeVandermonde( Kokkos::View<double const **, DeviceType> points,
+                        PolynomialBasis const &polynomial_basis )
+    {
+        auto const n_points = points.extent( 0 );
+        auto constexpr size_polynomial_basis = PolynomialBasis::size();
+        Kokkos::View<double *, DeviceType> p(
+            "vandermonde", n_points * size_polynomial_basis );
```

Comment:
If all points have the same basis size why is this view 1D instead of 2D?

Reply:
I think at some point we were thinking about batched interface, which would most likely take in 1D views.